pkg/scheduler/reconciliation/persistent.go

```diff
@@ -618,3 +647,21 @@ func (r *PersistentReconciliationRepository) GetAllComponents() ([]string, error
 
 	return components, nil
 }
+
+func splitStringSlice(slice []string, blockSize int) [][]string {
+	count := len(slice)
+	if len(slice) == 0 {
+		return nil
+	}
+	var blocks [][]string
+	var j int
+	for i := 0; i < count; i += blockSize {
+		j += blockSize
+		if j > count {
+			j = count
+		}
+
+		blocks = append(blocks, slice[i:j])
+	}
+	return blocks
+}
```

Comment:
just a small thing here, feel free to leave as is if you think its more readable :) 
```suggestion
func chunkSlice(slice []string, chunkSize int) [][]string {
	if len(slice) == 0 {
		return nil
	}

	// initialize with final size for alloc
	chunks := make([][]string, (len(slice)+chunkSize-1)/chunkSize)

	i, left, limit := 0, 0, len(slice)-chunkSize

	for left < limit {
		right := left + chunkSize
		chunks[i], left = slice[left:right], right
		i++
	}
	chunks[i] = slice[left:]

	return chunks
}
```
```

Reply:
Thanks for the suggestion. I have adapted my function to make it more readable and incorporated returned slice capacity. Inline calculations (like ceiling above), in my opinion are efficient, but not intuitive.